tests/src/checkfebyautodiff.hh

```diff
@@ -80,13 +80,14 @@ auto checkFESByAutoDiffImpl(const GridView& gridView, const BasisHandler& basis,
 
 template <typename GridView, typename PreBasis, typename Skills, typename AffordanceColl>
 auto checkFESByAutoDiff(const GridView& gridView, const PreBasis& pb, Skills&& skills, AffordanceColl affordance,
-                        const std::string& testName = "") {
+                        const std::string& testName = "", double tol = 1e-10) {
   Dune::TestSuite t("AutoDiff Test" + testName);
   auto basis = Ikarus::makeBasis(gridView, pb);
   Eigen::VectorXd d;
   d.setZero(basis.flat().dimension());
-  t.subTest(checkFESByAutoDiffImpl(gridView, basis, skills, affordance, d, " Zero Displacements"));
+  t.subTest(checkFESByAutoDiffImpl(gridView, basis, skills, affordance, d, " Zero Displacements", tol));
   d.setRandom(basis.flat().dimension());
-  t.subTest(checkFESByAutoDiffImpl(gridView, basis, skills, affordance, d, " Non-zero Displacements"));
+  d *= 0.2; // to avoid a negative determinant of deformation gradient
```

Comment:
Hmm maybe there is a more portable way to do this. If we change the single element creation from 0...1 to something more general that we can change it to `lower` and `upper` then this breaks again. So the displacements should just be a fraction of the size of the coordinates.

Reply:
I would propose to move this to a separate PR, if at all we change the creation of the grid within the test suite (perhaps while handling #211). If the test breaks again, at least we have a proper error message now that would point us directly to its source.

Reply:
Yes thats fine.